libs/datamodel/core/src/configuration/source/datasource.rs

```diff
@@ -0,0 +1,26 @@
+use crate::StringFromEnvVar;
+use datamodel_connector::Connector;
+
+pub struct Datasource {
+    pub name: String,
+    pub provider: Vec<String>,
+    pub active_provider: String,
+    pub url: StringFromEnvVar,
+    pub documentation: Option<String>,
+    // a connector representing the intersection of all providers specified
+    pub combined_connector: Box<dyn Connector>,
+    pub active_connector: Box<dyn Connector>,
+}
+
+impl Datasource {
+    pub fn url(&self) -> &StringFromEnvVar {
+        &self.url
+    }
+
+    pub fn set_url(&mut self, url: &str) {
+        self.url = StringFromEnvVar {
+            from_env_var: None,
+            value: url.to_string(),
+        };
+    }
```

Comment:
I don't quite follow what's going on in here; could we add documentation?

Reply:
What would you add here? Don't know what to document in this case. I added some documentation to `StringFromEnvVar` in the followup PR. Check this one out if it helps.

Reply:
> What would you add here?

Haa, trick question! If I knew I would tell you.

I guess I'm unsure in general how `StringFromEnvVar` relates to the rest of the system. Judging by name I'm wondering why it's not just a `String`. Some pointers on what the *intent* of this construct would be helpful for anyone trying to understand what's going on here.